pkg/arvo/sys/arvo.hoon

```diff
@@ -417,29 +417,67 @@
 ++  de
   =|  fat=(axal)
   |@
+  ++  del
+    |=  pax=path
+    ?~  pax  [~ dir.fat]
+    =/  kid  (~(get by dir.fat) i.pax)
+    ?~  kid  fat
+    $(fat u.kid, pax t.pax)
+  ::  Descend to the axal at this path
   ::
-  ++  get
+  ++  dip
     |=  pax=path
     ^+  fat
     ?~  pax  fat
     =/  kid  (~(get by dir.fat) i.pax)
     ?~  kid  [~ ~]
     $(fat u.kid, pax t.pax)
   ::
-  ++  put
-    |*  [pax=path dat=*]
-    =>  .(dat `_?>(?=(^ fil.fat) u.fil.fat)`dat)
-    ^+  fat
-    ?~  pax  fat(fil `dat)
-    =/  kid  (~(get by dir.fat) i.pax)
-    =/  new  (fall kid fat(fil ~, dir ~))
-    fat(dir (~(put by dir.fat) i.pax $(fat new, pax t.pax)))
-  ::
   ++  gas
     |=  lit=(list (pair path _?>(?=(^ fil.fat) u.fil.fat)))
     ^+  fat
     ?~  lit  fat
     $(fat (put p.i.lit q.i.lit), lit t.lit)
+  ::  Fetch file at longest existing prefix of the path
+  ::
+  ++  get
+    |=  pax=path
+    ^+  [path fat]
+    ?~  pax  [~ fil.fat]
+    =/  kid  (~(get by dir.fat) i.pax)
+    ?~  kid  [pax fil.fat]
+    =/  low  $(fat u.kid, pax t.pax)
+    ?~  +.low
+      [pax fil.fat]
+    low
```

Comment:
This may be *the iconic trie operation*, but I think it's worth considering that all our call sites use `+dip` instead. I'd be inclined to swap the names based on that alone (I realize this was hashed out elsewhere).

Reply:
I think of `get` as getting an in item in the container, not returning an instance of the container itself, which is why `+an:cloy` and lib/trie used that for `+get`.  Something like `(~(get xx (~(put xx foo) bar baz)))` should generally be `baz`.  I guess `+get:to` is a partial counterexample, but it's still focused on the item, not the container.

Our +dip call-sites are mostly (5 out of the 8) referring to the item anyway, which is why +get feels natural for them.  I agree that it's strange to give the standard retrieval name to the "loose" get.  So maybe we really should have three:

- `+dip` descends into the axal, producing a new axal
- `+get` produces the item at that path, producing a unit item
- `+fit` produces the item at the longest existing prefix to the path, producing `[remaining-path (unit item)]`

I think that will be the least surprising interface